processing/src/main/java/io/druid/segment/ColumnSelectorFactory.java

```diff
@@ -31,5 +31,6 @@
   public FloatColumnSelector makeFloatColumnSelector(String columnName);
   public LongColumnSelector makeLongColumnSelector(String columnName);
   public ObjectColumnSelector makeObjectColumnSelector(String columnName);
+  public NumericColumnSelector makeMathExpressionSelector(String expression);
```

Comment:
this could take `Expr` to avoid parsing overhead for each segment's query runner.